std/encoding.d

```diff
@@ -619,27 +628,7 @@ struct CodeUnits(E)
         s = encode!(E)(d);
     }
 
-    int opApply(scope int delegate(ref E) dg)
-    {
-        int result = 0;
-        foreach(E c;s)
-        {
-            result = dg(c);
-            if (result != 0) break;
-        }
-        return result;
-    }
-
-    int opApplyReverse(scope int delegate(ref E) dg)
-    {
-        int result = 0;
-        foreach_reverse(E c;s)
-        {
-            result = dg(c);
-            if (result != 0) break;
-        }
-        return result;
-    }
+    alias s this;
```

Comment:
At that point, just return a slice.

The current `CodeUnits` only provides `foreach` functionality. With this AliasThis, `CodeUnits` provides *all* functionality available to slices, including reassignment, shrinking, appending, etc.